src/h.js

```diff
@@ -1,22 +1,29 @@
-export function h(tag, data) {
-  var node
-  var stack = []
-  var children = []
+function addChild(children, val) {
+  if (val == null || typeof val === 'boolean') return
 
-  for (var i = arguments.length; i-- > 2; ) {
-    stack[stack.length] = arguments[i]
+  if (Array.isArray(val)) {
+    for (var j = 0; j < val.length; j++) {
```

Comment:
@ngryman That's some great info!!

I don't really have anything constructive to add here, because everyone gave a lot of objective points.  But, I'll just say I've noticed this redundancy in my own benchmarks.  Put simply:  I don't think caching the length is necessary because this type of loop is not uncommon, and it would make sense to me for V8 to optimize this sort of thing instead of forcing everyone to cache their lengths to get a perf gain.

Does this have other effects on other engines?  Curious how this performs cross-browser.